### PR TITLE
Bugfix/no repeats in batch again

### PR DESCRIPTION
**Pull request by kroenlein**
# Citrine Python PR

## Description 
This restores the batching corrections from #747 and updates register_all calls to go to the storables route.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [X] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [X] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/resources/data_concepts.py [hunk 1]
```diff
@@ -544,73 +542,14 @@ def register_all(self,
...
-                              lambda x: x.uids.pop(temp_scope, None))  # Strip temp uids
-        return resources
+        from citrine.resources.gemd_resource import GEMDResourceCollection
+        gemd_collection = GEMDResourceCollection(self.project_id, self.dataset_id, self.session)
+        return gemd_collection.register_all(
+            models,
+            dry_run=dry_run,
+            status_bar=status_bar,
+            include_nested=include_nested
+        )
```
Comment (kroenlein) [inline src/citrine/resources/data_concepts.py:552]: All DataConcepts register_all calls should route through GEMDResourceCollection / `storables`

File: src/citrine/resources/dataset.py [hunk 2]
```diff
@@ -197,7 +197,7 @@ def files(self) -> FileCollection:
 
     def register(self, model: DataConcepts, *, dry_run=False) -> DataConcepts:
         """Register a data model object to the appropriate collection."""
-        return self.gemd.register(model, dry_run=dry_run)
+        return self.gemd._collection_for(model).register(model, dry_run=dry_run)
```
Comment (kroenlein) [inline src/citrine/resources/dataset.py:200]: All dataset register calls should route through the particular object

File: src/citrine/resources/dataset.py [hunk 3]
```diff
@@ -250,7 +250,7 @@ def register_all(self,
 
     def update(self, model: DataConcepts) -> DataConcepts:
         """Update a data model object using the appropriate collection."""
-        return self.gemd.update(model)
+        return self.gemd._collection_for(model).update(model)
```
Comment (kroenlein) [inline src/citrine/resources/dataset.py:253]: All DataConcepts update calls should route through the particular collection.

File: src/citrine/resources/dataset.py [hunk 4]
```diff
@@ -265,7 +265,11 @@ def delete(self, uid: Union[UUID, str, LinkByUID, DataConcepts], *, dry_run=Fals
             Dry run is intended to be used for validation. Default: false
 
         """
-        return self.gemd.delete(uid, dry_run=dry_run)
+        if isinstance(uid, DataConcepts):
+            collection = self.gemd._collection_for(uid)
+        else:
+            collection = self.gemd
+        return collection.delete(uid, dry_run=dry_run)
```
Comment (kroenlein) [inline src/citrine/resources/dataset.py:272]: dataset.delete will route through the object route if we have a type, and through `storables` if not.

File: src/citrine/resources/gemd_resource.py [hunk 5]
```diff
@@ -50,30 +60,6 @@ def build(self, data: dict) -> DataConcepts:
...
-            Dry run is intended to be used for validation. Default: false
-
-        """
-        model = self.get(uid)  # Get full object for collection lookup
-        return self._collection_for(model).delete(model, dry_run=dry_run)
-
-    def register(self, model: DataConcepts, *, dry_run=False) -> DataConcepts:
-        """Register a GEMD object to the appropriate collection."""
-        return self._collection_for(model).register(model, dry_run=dry_run)
-
```
Comment (kroenlein) [inline src/citrine/resources/gemd_resource.py:76]: These are now naturally dispatched through DataConcepts.  You get a `storables` route iff you ask for it.

File: tests/resources/test_dataset.py [hunk 6]
```diff
@@ -349,6 +350,11 @@ def test_gemd_posts(dataset):
         dataset.delete(updated)
         assert dataset.session.calls[-1].path.split("/")[-3] == basename(collection._path_template)
 
+        # But they would have dispatched differently w/ just a uid
+        dataset.session.set_response(updated.dump())
+        dataset.delete(updated.uid)
+        assert dataset.session.calls[-1].path.split("/")[-3] == basename(GEMDResourceCollection._path_template)
+
```
Comment (kroenlein) [inline tests/resources/test_dataset.py:357]: Test uid v. object branching.  Not actually registered, so repeat delete isn't a problem.

File: tests/resources/test_material_run.py [hunk 7]
```diff
@@ -57,8 +58,8 @@ def test_register_all(collection, session):
     assert [r.name for r in runs] == [r.name for r in registered]
     assert len(session.calls) == 1
     assert session.calls[0].method == 'PUT'
-    assert session.calls[0].path == 'projects/{}/datasets/{}/material-runs/batch'.format(
-        collection.project_id, collection.dataset_id)
+    assert GEMDResourceCollection(collection.project_id, collection.dataset_id, collection.session)._get_path() \
+           in session.calls[0].path
```
Comment (kroenlein) [inline tests/resources/test_material_run.py:62]: Even though you asked for `material_run`, it was a `register_all` so it dispatched to `storables`.

**Comment by kroenlein** [comment]
I can verify functionality against stage & dev, but I don't have tests against local-devkit configured.

**Comment by kroenlein** [comment]
end-to-end tests:
https://jenkins.corp.citrine.io/job/nextgen-ITs/26519
https://jenkins.corp.citrine.io/job/nextgen-ITs/26520

**Comment by kroenlein** [comment]
https://jenkins.corp.citrine.io/job/nextgen-ITs/27027/
Tests all pass.

File: src/citrine/resources/dataset.py (hunk 3 above)
Comment (djack) [reply src/citrine/resources/dataset.py:253]: why?

File: src/citrine/resources/dataset.py (hunk 4 above)
Comment (djack) [reply src/citrine/resources/dataset.py:272]: why?

**Review by djack — APPROVED** [review]
passes tests, improves the state of the world.

**Pull request merged**